routing/pathfind.go

```diff
@@ -542,14 +542,19 @@ func findPath(g *graphParams, r *RestrictParams, cfg *PathFindingConfig,
 		// search is taken place from target to source.
 		for _, outgoingEdgePolicy := range outgoingEdgePolicies {
 			toVertex := outgoingEdgePolicy.ToNodePubKey()
```

Comment:
This only needs to be executed if `vertex != self`

Reply:
[6c94e09](https://github.com/lightningnetwork/lnd/pull/6766/commits/6c94e09a6f2b183dda4bd0d6ffe63fdf6a8ae618)